src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisTable.tsx

```diff
@@ -0,0 +1,318 @@
+              time: curDay,
+              userId: userOverviewUsage.userId,
+            })
+          }
+        }
+      });
+    }
+
+    if (this.filterToggled && data.length > 0 ) {
+      let tableFormat: string;
+      if (this.timeTypeToggleValue === ToggleValue.RESULTS_BY_MONTH) {
+        tableFormat = TABLE_MONTH_FORMAT;
+      } else if (this.timeTypeToggleValue === ToggleValue.RESULTS_BY_DAY) {
+        tableFormat = TABLE_DAY_FORMAT;
+      }
+      data = data.filter(resource => {
+        const fromTime = moment(this.fromDate).format(tableFormat);
+        const toTime = moment(this.toDate).format(tableFormat);
+        return resource.time >= fromTime && resource.time <= toTime;
+      });
+    }
+
+    if (this.resourcesTypeToggleValue === ToggleValue.ALL_RESOURCES) {
+      return data;
+    } else if (this.resourcesTypeToggleValue === ToggleValue.PUBLIC_RESOURCES) {
+      return (
+        _.filter(data, function (usage) {
+          return !usage.resource.includes('/private/');
+        })
+      );
+    } else {
+        return [];
+    }
+  }
+
+  @computed get resetDefaultSort(): SortingRule[] {
+    const sortingRules: SortingRule[] = [];
+    if (this.filterToggled) {
+      sortingRules.push({
+        id: UsageTableColumnKey.TIME,
+        desc: false,
+      });
+    } else {
+      sortingRules.push({
+        id: UsageTableColumnKey.TIME,
+        desc: true,
+      });
+    }
+
+    if (this.timeTypeToggleValue === ToggleValue.RESULTS_IN_TOTAL) {
+      sortingRules.push({
+        id: 'totalUsage',
+        desc: true,
+      });
+    } else if (this.timeTypeToggleValue === ToggleValue.RESULTS_BY_MONTH) {
+      sortingRules.push({
+          id: UsageTableColumnKey.USAGE,
+          desc: true,
+        },
+        {
+          id: 'monthResetPlaceholder',
+          desc: true,
+      });
+    } else if (this.timeTypeToggleValue === ToggleValue.RESULTS_BY_DAY) {
+      sortingRules.push({
+          id: UsageTableColumnKey.USAGE,
+          desc: true,
+        },
+        {
+          id: 'dayResetPlaceholder',
+          desc: true,
+        });
+    }
+
+    return sortingRules;
+  }
+
+  render() {
+    return (
+      <>
+        <OncoKBTable<UserOverviewUsage | UsageRecord>
+          data={this.calculateData}
+          columns=
+            {this.timeTypeToggleValue === ToggleValue.RESULTS_IN_TOTAL
+            ? [{
+                id: 'userEmail',
+                Header: emailHeader,
+                accessor: 'userEmail',
+                onFilter: (row: UserOverviewUsage, keyword: string) =>
+                  filterByKeyword(row.userEmail, keyword),
+              },
+                {
+                  id: 'totalUsage',
+                  Header: usageHeader,
+                  accessor: 'totalUsage',
+                  Cell(props: { original: UserOverviewUsage }) {
+                    return <UsageText usage={props.original.totalUsage ? props.original.totalUsage : 0} />;
```

Comment:
I think as a primitive data type, they all have default values. Why we are checking here?